react/week-22/lesson.md

```diff
@@ -0,0 +1,120 @@
+# React 4
+
+![Lesson Ready](https://img.shields.io/badge/status-ready-green.svg)
+
+
+## Introduction to React-Router
+
+In the past few weeks, you've learned how to build applications with React and the different applications and examples were all built on a single page. However, what if you wanted to have different pages with each page having its own URL (so you can bookmark it for example)? You will need to introduce a router in your application. In JavaScript, a router is the piece of code which is in charge of switching between views of your application and keep each view in sync with a specific URL. For example, you could imagine having a homepage reachable from the root path `/` and a users page with the path `/users`. In React, a popular library to help you achieve this is React-Router.
+
+Let's look at a first example ([interactive example](https://codesandbox.io/s/react-router-1-lk44e)):
+
+```js
+const Index = () => {
+  return <h2>Home</h2>;
+};
+
+const About = () => {
+  return <h2>About</h2>;
+};
+
+const Users = ({ names }) => {
+  return (
+    <div>
+      <h2>Users</h2>
+      <ul>
+        {names.map(name => <li>{name}</li>)}
+      </ul>
+    </div>
+  );
+};
+
+const AppRouter = () => {
+  return (
+    <Router>
+      <div>
+        <nav>
+          <ul>
+            <li>
+              <Link to="/">Home</Link>
+            </li>
+            <li>
+              <Link to="/about/">About</Link>
+            </li>
+            <li>
+              <Link to="/users/">Users</Link>
+            </li>
+          </ul>
+        </nav>
+
+        <Route path="/" exact component={Index} />
```

Comment:
Is it worth introducing this with `<Switch>` - the exclusive behaviour is probably less likely to cause surprises down the line than the inclusive behaviour.

Reply:
That's a fair point, I wasn't sure but I tried finally to follow what is introduced on the official website of React Router in the quick start and avoid to introduce too many things at the same time. I think students can probably try `Switch` as a homework or further reading exercise? What do you think?

Reply:
@textbook I'm going to merge this for now but happy to discuss it again.